source/frontend/src/components/maps/MapLeafletView.tsx

```diff
@@ -150,12 +154,20 @@ const MapLeafletView: React.FC<React.PropsWithChildren<MapLeafletViewProps>> = (
         mapRef?.current?.flyTo(requestedFlyTo.location, MAP_MAX_ZOOM, {
           animate: false,
         });
+        mapRef?.current?.panTo(requestedFlyTo.location);
       }
 
       mapMachineProcessFlyTo();
     }
   }, [isMapReady, hasPendingFlyTo, requestedFlyTo, mapMachineProcessFlyTo]);
 
+  useEffect(() => {
+    if (hasPendingCenterTo && isMapReady && requestedCenterTo.location) {
+      mapRef.current?.setView(requestedCenterTo.location);
+    }
+    mapMachineProcessCenterTo();
+  }, [hasPendingCenterTo, isMapReady, mapMachineProcessCenterTo, requestedCenterTo.location]);
```

Comment:
Is the setview not the same as flyto without the animation?

Reply:
They are similar in the sense they both center the map to a location, but 'FlyTo' is better suited for changing the map location while doing pan-zoom. Based on the documentation 'SetView' is better suited for this case.

Reply:
I don't need to modify the current zoom level for the new requirement. Additionally there's no easy way to use the 'flyTo' method to only center as per the reequipment. It would be more risky to modify the current 'flyTo' method to  calculate the current bounds.